InfoMovies/InfoMovies/ViewModels/MainPageViewModel.cs

```diff
@@ -0,0 +1,163 @@
+        private ObservableRangeCollection<MovieViewModel> movies;
+
+        #endregion
+
+        #region Properties
+
+        public ObservableRangeCollection<MovieViewModel> Movies
+        {
+            get => movies;
+            set => SetProperty(ref movies, value);
+        }
+
+        public bool IsRefreshing
+        {
+            get => isRefreshing;
+            set => SetProperty(ref isRefreshing, value);
+        }
+
+        public bool IsSearching
+        {
+            get => isSearching;
+            set => SetProperty(ref isSearching, value);
+        }
+
+        public string SearchText
+        {
+            get => searchText;
+            set => SetProperty(ref searchText, value, onChanged:OnSerachTextUpdate);
+        }
+
+        public ICommand SearchCommand { get; private set; }
+
+        public ICommand RefreshCommand { get; private set; }
+
+        #endregion
+
+        #region Constructors
+
+        public MainPageViewModel()
+        {
+            Movies = new ObservableRangeCollection<MovieViewModel>();
+
+            SearchCommand = new Command<string>(async (textParameter) =>
+                await SearchCommandExecute(textParameter));
+
+            RefreshCommand = new Command(async () =>
+            {
+                IsRefreshing = false;
+                if (IsBusy)
+                    return;
+
+                currentPage = 1;
+                movies.Clear();
+
+                await LoadItemsAsync();
+            });
+        }
+        #endregion
+
+        #region Public Methods
+
+        public async Task LoadItemsAsync()
+        {
+            if (IsBusy)
+                return;
+
+            IsBusy = true;
+            isSearching = false;
+            SearchText = string.Empty;
+
+            var tmdbMovies = await _provider.GetUpcommingMoviesAsync(currentPage);
+            if (!(tmdbMovies?.Count() > 0))
+            {
+                IsBusy = false;
+                return;
+            }
+            currentPage++;
+            FillMoviesCollection(tmdbMovies);
+            IsBusy = false;
+        }
+
+        public async Task LoadSearchItemsAsync()
+        {
+            await SearchCommandExecute(SearchText);
+        }
+
+        #endregion
+
+        private async Task SearchCommandExecute(string searchText)
+        {
+            IsBusy = true;
+
+            if (string.IsNullOrEmpty(searchText))
+            {
+                if(isSearching)
+                {
+                    movies.Clear();
+                    await LoadItemsAsync();
+                }   
+            }
+            else
+            {
+                if (!isSearching)
+                {
+                    currentPage = 1;
+                    movies.Clear();
+                }
+
+                isSearching = true;
+
+                var tmdbMovies =  await _provider.SearchAsync(searchText, currentPage);
+                FillMoviesCollection(tmdbMovies);
```

Comment:
What if the tmdbMovies list is null? As you have handled to return null every time that something goes wrong, you should care about handling null objects when making requests. How can you improve the error handling especially for web requests?

Reply:
i skipped that, my bad... D:
We can replace `null` returns to  `Default()`

Reply:
Returning default() here would cause the same problem. The default value for a list is null.

Reply:
yeah! thats true.. sorry, i don't make me clear..
i was thinking about web requests (generics), to return default(T)
Like `Web Api Client`
But talking about collections x webrequest, we can crate a default collection and fill that with return of webrequest, if not null.
So, we can return this collection with web request return or a collection with none items